sdks/go/pkg/beam/core/runtime/exec/pardo.go

```diff
@@ -162,6 +168,14 @@ func (n *ParDo) processMainInput(mainIn *MainInput) error {
 	return nil
 }
 
+func rtErrHelper(err error) error {
+	if err != nil {
+		return err
+	} else {
+		return errors.New("RTracker IsDone failed for unspecified reason")
+	}
```

Comment:
```suggestion
	}
	return errors.New("RTracker.IsDone() failed for unspecified reason")
```

Reply:
Also, in both places this is called, error or no, we don't know IsDone failed(), we know that the DoFn terminated without fully processing the restriction (If I understand IsDone's semantics correctly). Perhaps the message should indicate that?

Reply:
Yeah good point. I'll rephrase it.